### ISSUE TITLE
Still going to sleep - running Fedora 36

### ISSUE THREAD

**Issue by noelgarber** [fohrloop/wakepy#18]
I'm running Fedora 36 on a Lenovo Legion 7, and I ran the simple code: 

`from wakepy import set_keepawake, unset_keepawake`
`set_keepawake()`

Yet my computer still goes to sleep after a few minutes. What's happening?

**Comment by np-8** [comment]
Hi @noelgarber and thanks for filing the issue!

I think we could check a two things here:

#### 1) systemctl
- As mentioned in the [docs](https://github.com/np-8/wakepy), wakepy uses `systemctl mask` to prevent computer going to sleep, when used on Linux. Does your system have `systemd`?  If yes, try to run

```
systemctl mask sleep.target suspend.target hibernate.target hybrid-sleep.target
```
and see if that prevents your computer from going to sleep or not

#### 2) privileges

The `systemctl` command needs sudo privileges. Because current version of wakepy uses `systemctl`, it needs sudo. Did you run you script with sudo privileges?

**Comment by noelgarber** [comment]
Thanks for the reply! Here are the answers: 

1) Yes, I do have systemd (as verified by `ps -p 1 -o comm=`), and no, that command does not prevent my computer from going to sleep - it goes to sleep after 5 minutes even after running that. Here is what came up in terminal: 

`[ngarber@legion7 ~]$ systemctl mask sleep.target suspend.target hibernate.target hybrid-sleep.target`
`Created symlink /etc/systemd/system/sleep.target → /dev/null.`
`Created symlink /etc/systemd/system/suspend.target → /dev/null.`
`Created symlink /etc/systemd/system/hibernate.target → /dev/null.`
`Created symlink /etc/systemd/system/hybrid-sleep.target → /dev/null.`

After typing the command, it prompts for a password before executing, which I entered. Same behaviour as when I run set_keepawake() from wakepy. 

2) It asks for a password when I run set_keepawake() from wakepy, so I assume entering the password gives the necessary permissions. 

Unfortunately, when I run `sudo python3` and then try to run `from wakepy import set_keepawake, unset keepawake`, it returns `ModuleNotFoundError: No module named 'wakepy'`. Somehow, when running inside sudo, Python can't find wakepy despite it being installed on my computer...

**Comment by np-8** [comment]
Hi and sorry for long delay!

For **part 2**, I think it is because sudo has different set of environment variables than your user, and wakepy is installed to a folder which is added (only) to the PATH of your current user. See, for example: [How to make `sudo` preserve $PATH?](https://unix.stackexchange.com/questions/83191/how-to-make-sudo-preserve-path). Based on that I guess that

```
sudo env "PATH=$PATH" python3
```

would preserve your PATH with sudo.

**Comment by np-8** [comment]
For **part 1**, I found a least [this](https://unix.stackexchange.com/a/228461/92705) answer which has a comment which says that commands like 

```
sudo systemctl mask suspend.target
sudo systemctl mask sleep.target
```

would *not* work on Fedora. I tried to search for a solution for Google but could not find much of anything useful. One solution at [ask.fedoraproject.org/](https://ask.fedoraproject.org/t/how-do-i-disable-automatic-suspend-in-fedora-36/29042/6) adviced to install `xset` and run

```
xset s off
```
Looking at [wiki.archlinux.org](https://wiki.archlinux.org/title/Display_Power_Management_Signaling) it seems that the command is merely "Disable screen saver blanking", and 

```
xset s off -dpms
```

would be "Disable DPMS and prevent screen from blanking", where DPMS = Display Power Management Signaling. Not sure if that only applies to the _display_, though. At the same page it also mentions [ `setterm`](https://man7.org/linux/man-pages/man1/setterm.1.html).  

Let me hear if you (or anyone else) found a solution for Fedora :)

**Comment by np-8** [comment]
@noelgarber I just published wakepy 0.6.0 and it has two options of using DBus + inhibit method. I suppose this will fix also your problems on Fedora!

**Comment by fohrloop** [comment]
I haven't checked if this is fixed yet but added 0.8.0 milestone, as Fedora (Workstation 36) uses GNOME 42, and support for GNOME using org.gnome.SessionManager (D-Bus) is in the [list of changes for 0.8.0](https://github.com/fohrloop/wakepy/discussions/37).

**Comment by fohrloop** [comment]
Just installed Fedora 36 for testing purposes. Running with the latest `dev` version:

```python
In [1]: import wakepy

In [2]: wakepy.__version__
Out[2]: '0.8.0dev'

In [3]: from wakepy import keep

In [4]: import time
   ...: with keep.running() as m:
   ...:     print(m.active)
   ...:     print(m.activation_result.active_method)
   ...:     time.sleep(2000)
```

Confirming that the org.gnome.SessionManager based keep.running & keep.presenting methods and the org.freedesktop.ScreenSaver based keep.presenting method work well on Fedora 36 with GNOME 42.0, using the latest 0.8.0dev.
